app/src/main/java/com/compose/quotes_jetpack/MainActivity2.kt

```diff
@@ -1,29 +1,116 @@
 package com.compose.quotes_jetpack
 
+import Models.DataManager
+import Models.QuotesData
+import Models.favouritQuote
+import Screens.QuoteMain
+import android.annotation.SuppressLint
 import android.os.Bundle
+import android.widget.Toast
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
-import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.Card
 import androidx.compose.material.MaterialTheme
 import androidx.compose.material.Surface
 import androidx.compose.material.Text
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
 import com.compose.quotes_jetpack.ui.theme.Quotes_JetpackTheme
+import com.google.firebase.firestore.ktx.firestore
+import com.google.firebase.firestore.ktx.toObject
+import com.google.firebase.ktx.Firebase
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
+import java.lang.StringBuilder
 
 class MainActivity2 : ComponentActivity() {
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
-
-        }
+       GetFromFirestore()
     }
 }
+    @Composable
```

Comment:
First we created a state variable which would be initialize with a empty list and its a mutable state variable that is required to hold the data from firestore. Now here remember function is to remember the state across recomposition meaning preserving the value of state variable during the recomposition occurence so that the state variable is not again initalized and hold its initial value.
Now the LaunchedEffect is a compose effect used to launch a coroutine and run the code inside the block.
Now just the LaunchedEffect block consist of the code for getting data from the Firestore database and adding it to a list.
After that we have written a line which is data.value=datalist which means that we are updating the value of the state variable with the data fetched from the firestore database.
The importance of doing this is that as the value of the state changes it triggers recomposition in compose which would refresh the UI and reflect the updated Data.